x/oracle/types/oracle.go

```diff
@@ -13,19 +14,73 @@ func (m SignedOracleRegistrationVote) ValidateBasic() error {
 	if len(m.OracleRegistrationVote.UniqueId) == 0 {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "uniqueID is empty")
 	}
+
 	if len(m.OracleRegistrationVote.UniqueId) == 0 {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "uniqueID is empty")
 	}
-	_, err := sdk.AccAddressFromBech32(m.OracleRegistrationVote.VoterAddress)
-	if err != nil {
+
+	if _, err := sdk.AccAddressFromBech32(m.OracleRegistrationVote.VoterAddress); err != nil {
 		return err
 	}
-	_, err = sdk.AccAddressFromBech32(m.OracleRegistrationVote.VotingTargetAddress)
-	if err != nil {
+
+	if _, err := sdk.AccAddressFromBech32(m.OracleRegistrationVote.VotingTargetAddress); err != nil {
 		return err
 	}
+
 	if len(m.OracleRegistrationVote.EncryptedOraclePrivKey) == 0 {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "encryptedOraclePrivKey is empty")
 	}
 	return nil
 }
+
+func (m Oracle) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Address); err != nil {
+		return sdkerrors.Wrapf(err, "oracle address is invalid. address: %s", m.Address)
+	}
+	return nil
+}
+
+func (m OracleRegistration) ValidateBasic() error {
+	if len(m.UniqueId) == 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "uniqueID is empty")
+	}
+	if _, err := sdk.AccAddressFromBech32(m.Address); err != nil {
+		return sdkerrors.Wrapf(err, "oracle address is invalid. address: %s", m.Address)
+	}
+
+	if m.NodePubKey != nil {
+		if _, err := btcec.ParsePubKey(m.NodePubKey, btcec.S256()); err != nil {
+			return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "nodePubKey is invalid. %s", err.Error())
+		}
+	}
+
+	if m.TrustedBlockHeight <= 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "trustedBlockHeight must be greater than zero")
+	}
+
+	if m.TrustedBlockHash == nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "trustedBlockHash is nil")
+	}
+
+	return nil
+}
+
+func (m OracleRegistrationVote) ValidateBasic() error {
```

Comment:
`m.VoteOption` should be one of the enum type of `VoteOption`.

Reply:
Yes, I added validation before [PR](https://github.com/medibloc/panacea-core/pull/364/files#diff-34aa5cd30b13359bc81fc82388624970b489c973ce0b85a7982beb48a4e96b77)